torch/linalg/__init__.py

```diff
@@ -272,6 +272,92 @@
             [-3.1113,  2.7381]], dtype=torch.float64)
 """)
 
+lstsq = _add_docstr(_linalg.linalg_lstsq, r"""
+torch.linalg.lstsq(input, b, cond=None, driver_name=None) -> (Tensor x, Tensor rank, Tensor s)
+
+Computes the least squares solution to the system with a batch of matrices :math:`a` (represented by :attr:`input`)
+and a batch of vectors/matrices :math:`b` such that
+
+.. math::
+    x = \text{argmin}_x \|ax - b\|_F,
+
+where :math:`a` is of size :math:`(..., m, n)` and :math:`b` is of size :math:`(..., m, k)` or
+:math:`(..., m)`. The batch dimensions of :math:`a` and :math:`b` have to be broadcastable.
+
+The returned solution :math:`solution` is of shape :math:`(..., \max(m, n), k)` if :math:`b` is of shape :math:`(..., m, k)`,
+and is of shape :math:`(..., \max(m, n), 1)` if :math:`b` is of shape :math:`(..., m)`.
+The batch sizes of :math:`x` is the broadcasted shape of the batch dimensions of :math:`a` and :math:`b`.
+The minimizer could be obtained by slicing the solution with, for example, ``x.narrow(-2, 0, n)``.
+The remaining entries ``x.narrow(-2, n, max(m, n) - n)``
+encode the residual sum of squares for the solution in each column for full-rank matrices in :math:`a`,
+i.e. if :math:`\text{rank}(a[..., :, :]) = \min(m, n)`, the residuals for the system :math:`a[..., :, :]` could be obtained by
+``residuals = x.narrow(-2, n, max(m, n) - n).pow(2).sum(-2)[..., :, :]``.
```

Comment:
@rgommers , @mruberry , this is what I decided to do when it comes to handling residuals.